sdk/src/network-client.ts

```diff
@@ -1198,24 +1198,53 @@ class AleoNetworkClient {
    * const transaction = await networkClient.waitForTransactionConfirmation(transactionId);
    */
   async waitForTransactionConfirmation(
-      transactionId: string,
-      checkInterval: number = 2000,  // Poll every 2 seconds
-      timeout: number = 45000        // Timeout after 45 seconds
-  ): Promise<Transaction> {
+    transactionId: string,
+    checkInterval: number = 2000,
+    timeout: number = 45000
+  ): Promise<"accepted" | "rejected"> {
     const startTime = Date.now();
-
-    return new Promise<Transaction>((resolve, reject) => {
+  
+    return new Promise((resolve, reject) => {
       const interval = setInterval(async () => {
+        const elapsed = Date.now() - startTime;
+  
+        if (elapsed > timeout) {
+          clearInterval(interval);
+          return reject(new Error("Transaction confirmation timed out"));
+        }
+  
         try {
-          // Replace with actual Aleo transaction lookup API
-          const transaction = <Transaction>await this.getTransactionObject(transactionId);
-          resolve(transaction);
-          if (Date.now() - startTime > timeout) {
+          const res = await fetch(`https://api.explorer.provable.com/v1/mainnet/transaction/confirmed/${transactionId}`);
```

Comment:
We should be using the configured url.

```suggestion
const res = await get(this.host + `/transaction/confirmed/${transactionId}`, {
        headers: this.headers
      });
```

Reply:
noted - I've updated to use the configured URL. However, I am still using `fetch` rather than `get` here, as our `get` method throws immediately on non-OK responses, which prevents us from inspecting the response body. Let me know if you think this is an issue.